### PR TITLE
[fix][test]Flaky test testMaxPendingChunkMessages

### PR DESCRIPTION
**Pull request by liangyepianzhou**
Fixes https://github.com/apache/pulsar/actions/runs/5934883964/job/16092673164?pr=21043

### Motivation

Fix the flaky test.
### Modifications

The root cause is `doAcknowledge` is an async method.
The chunk 1 may not be acked when call `redeliverUnacknowledgedMessages`.
```
doAcknowledge(msgId, AckType.Individual, Collections.emptyMap(), null);
```
### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

**Comment by Technoboy-** [comment]
image

**Comment by liangyepianzhou** [comment]
> image

@Technoboy That is expected.

File: pulsar-broker/src/test/java/org/apache/pulsar/client/impl/MessageChunkingTest.java [hunk 1]
```diff
@@ -348,6 +363,10 @@ public void testMaxPendingChunkMessages() throws Exception {
         assertEquals(receivedMsg.getValue(), "chunk-1-0|chunk-1-1|");
 
         consumer.acknowledge(receivedMsg);
+        assertEquals(admin.topics().getStats(topicName).getSubscriptions().get(subName)
+                .getNonContiguousDeletedMessagesRanges(), 0);
```
Comment (RobertIndie) [inline pulsar-broker/src/test/java/org/apache/pulsar/client/impl/MessageChunkingTest.java:367]: Need to remove this line?

**Review by codelipenghui — APPROVED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/client/impl/MessageChunkingTest.java [hunk 2]
```diff
@@ -319,15 +320,29 @@ private void sendSingleChunk(Producer<String> producer, String uuid, int chunkId
...
+    /**
+     * This test used to test the consumer configuration of maxPendingChunkedMessage.
+     * If we set maxPendingChunkedMessage is 1 that means only one incomplete chunk message can be store in this
+     * consumer.
+     * For example:
+     * ChunkMessage1 chunk-1: uuid = 0, chunkId = 0, totalChunk = 2;
+     * ChunkMessage2 chunk-1: uuid = 1, chunkId = 0, totalChunk = 2;
+     * ChunkMessage2 chunk-2: uuid = 1, chunkId = 1, totalChunk = 2;
+     * ChunkMessage1 chunk-2: uuid = 0, chunkId = 1, totalChunk = 2;
+     * The chunk-1 in the ChunkMessage1 and ChunkMessage all is incomplete.
```
Comment (RobertIndie) [inline pulsar-broker/src/test/java/org/apache/pulsar/client/impl/MessageChunkingTest.java:332]:
```suggestion
     * The chunk-1 in the ChunkMessage1 and ChunkMessage2 all is incomplete.
```

File: pulsar-broker/src/test/java/org/apache/pulsar/client/impl/MessageChunkingTest.java (hunk 1 above)
Comment (liangyepianzhou) [reply pulsar-broker/src/test/java/org/apache/pulsar/client/impl/MessageChunkingTest.java:367]: This line is used to fix this flaky test.

Comment (codelipenghui) [reply pulsar-broker/src/test/java/org/apache/pulsar/client/impl/MessageChunkingTest.java:367]: https://github.com/apache/pulsar/pull/21103/files#diff-9ce7ee7d161127cb26c5c43598bc3cf93021123b69a5f8669c988bdb2a36b795R368-R369 will fixed the flaky test?

**Review by RobertIndie — APPROVED** [review]

**Pull request merged**